language/move-prover/tests/sources/defines.move

```diff
@@ -0,0 +1,19 @@
+module Test {
+
+    spec module {
+        define in_range(x: num, min: num, max: num): bool {
+            x >= min && x <= max
+        }
+
+        define eq<T>(x: T, y: T): bool {
```

Comment:
It is very cool that spec functions can be polymorphic as well.  I assume all the type checking entailed by type parameters is being done for spec functions also.  Great job!

Reply:
Yes, we are doing full type inference inclusive of overloading (so we can have len, all, any, etc. for different types if needed).